pyscf/neo/qc/qnuc.py

```diff
@@ -66,19 +68,50 @@ def dump_neo_qc_info(self, log):
     log.note("Hilbert-subspace-particle-conserving dimension: %g",fci_exc_tot)
     return
 
-def get_qubit_qnuc(mf_nuc):
-    '''returns (1) list of number of qubits associated with each quantum nucleus
-    (2) total number of quantum nuclear qubits
+def cas_selection(mf_nuc, cas_orb_nuc):
+    '''returns list nuc_coeff containing
+    MO matrices with orbitals according
+    to cas_orb_nuc specification
     '''
-    n_qubit_p = []
-    n_qubit_p_tot = 0
+    nuc_coeff = []
     for i in range(len(mf_nuc)):
-        n_qubit_p.append(mf_nuc[i].mo_coeff.shape[1])
-        n_qubit_p_tot += n_qubit_p[i]
-    return n_qubit_p, n_qubit_p_tot
+        coeff_mat = mf_nuc[i].mo_coeff[:, cas_orb_nuc]
+        nuc_coeff.append(coeff_mat)
+    return nuc_coeff
+
+def axis_finder(mol):
+    ''' simple code to find if we have a linear molecule
+        if so, report along what axis '''
+    from pyscf.hessian.thermo import rotation_const, _get_rotor_type
+    # find if this is a linear molecule
+    mass = mol.mass
+    atom_coords = mol.atom_coords()
+    mass_center = numpy.einsum('z,zx->x', mass, atom_coords) / mass.sum()
+    atom_coords = atom_coords - mass_center
+    rot_const = rotation_const(mass, atom_coords, 'GHz')
+    rotor_type = _get_rotor_type(rot_const)
+    if rotor_type == 'LINEAR':
+        # if a linear molecule, detect if the molecule is along x/y/z axis
+        if numpy.abs(atom_coords[:,0]).max() < 1e-6:
+            if numpy.abs(atom_coords[:,1]).max() < 1e-6:
+                axis = 2
+            elif numpy.abs(atom_coords[:,2]).max() < 1e-6:
+                axis = 1
+        elif numpy.abs(atom_coords[:,1]).max() < 1e-6:
+            if numpy.abs(atom_coords[:,2]).max() < 1e-6:
+                axis = 0
+        else:
+            # if not along an axis, warn
+            print('This is molecule is linear, but was not put along x/y/z axis. Symmetry will be OFF.',
```

Comment:
Avoid using print. Also, can you directly use symmetry_finder from neo.fci? You can add an extra layer of processing by detect if symm is LINEAR or not.